hoomd/hpmc/IntegratorHPMCMono.h

```diff
@@ -1804,6 +2088,907 @@ bool IntegratorHPMCMono<Shape>::py_test_overlap(unsigned int type_i, unsigned in
     return overlap;
     }
 
+/*! \param i The particle id in the list
+    \param pos_i Particle position being tested
+    \param shape_i Particle shape (including orientation) being tested
+    \param typ_i Type of the particle being tested
+    \param h_postype Pointer to GPUArray containing particle positions
+    \param h_orientation Pointer to GPUArray containing particle orientations
+    \param h_overlaps Pointer to GPUArray containing interaction matrix
+    \param hpmc_counters_t&  Pointer to current counters
+    \param hpmc_implicit_counters_t&  Pointer to current implicit counters
+    \param rng_depletants The RNG used within this algorithm
+    \param rng_i The RNG used for evaluating the Metropolis criterion
+
+    In order to determine whether or not moves are accepted, particle positions are checked against a randomly generated set of depletant positions.
+
+    NOTE: To avoid numerous acquires and releases of GPUArrays, data pointers are passed directly into this const function.
+    */
+#ifndef ENABLE_TBB
+template<class Shape>
+inline bool IntegratorHPMCMono<Shape>::checkDepletantOverlap(unsigned int i, vec3<Scalar> pos_i,
+    Shape shape_i, unsigned int typ_i, Scalar4 *h_postype, Scalar4 *h_orientation, unsigned int *h_overlaps,
+    hpmc_counters_t& counters, hpmc_implicit_counters_t *implicit_counters, hoomd::RandomGenerator& rng_depletants)
+#else
+template<class Shape>
+inline bool IntegratorHPMCMono<Shape>::checkDepletantOverlap(unsigned int i, vec3<Scalar> pos_i,
+    Shape shape_i, unsigned int typ_i, Scalar4 *h_postype, Scalar4 *h_orientation, unsigned int *h_overlaps,
+    hpmc_counters_t& counters, hpmc_implicit_counters_t *implicit_counters,
+    tbb::enumerable_thread_specific< hoomd::RandomGenerator >& rng_depletants_parallel)
+#endif
+    {
+    bool accept = true;
+
```

Comment:
Should we have an early exit when all fugacities are 0?

Reply:
it sort of already is an early exit. The only way we could 'speed this up' for many particle types is to cache the decision.

Reply:
however, we might then also skip the reduction of the counters. Not sure if that will be measurably faster, though.

Reply:
It will not likely improve performance, but it will make the code more readable.

Reply:
now fixed.